### ISSUE TITLE
UI Press - Build migration for Author content type

### ISSUE THREAD

**Issue by pyrello** [uiowa/uiowa#4663]
- 1249 items
- will be migrated into the Person content type

**Comment by pyrello** [comment]
We would like to figure out a way of identifying source images that are too small so that we can generate a list for fixing them later.

**Comment by cory-skeers** [comment]
> We would like to figure out a way of identifying source images that are too small so that we can generate a list for fixing them later.

Starting to look into this part of things... Only about a quarter of the files have dimensions data in the original database, so that only gets so far. I think the next steps would be to either look at the file data to pull out the dimensions and/or basing it on file size which can be misleading.

**Comment by pyrello** [comment]
> Starting to look into this part of things... Only about a quarter of the files have dimensions data in the original database, so that only gets so far. I think the next steps would be to either look at the file data to pull out the dimensions and/or basing it on file size which can be misleading.

I guess I was thinking pulling the dimensions from the file would be the way we would have to do it. I didn't realize that the dimensions got stored some of the time.

**Comment by cory-skeers** [comment]
> We would like to figure out a way of identifying source images that are too small so that we can generate a list for fixing them later.

Got this set up for the author photo, and currently set to 180x180 as the minimum (if either dimension is smaller, it won't be pulled over). Was a completely arbitrary decision on the minimum dimensions...Preferences on the minimum, @pyrello?  

Also, currently if it doesn't reach the minimum it won't be pulled in, but that can be updated if the idea was to keep the smaller image but list it as one to be replaced later. Realized I'd misread the original comment.

**Comment by joewhitsitt** [comment]
>Preferences on the minimum

Should it be >= to the smallest variant of the responsive image view mode that it is being migrated into?

**Comment by pyrello** [comment]
> Got this set up for the author photo, and currently set to 180x180 as the minimum (if either dimension is smaller, it won't be pulled over). Was a completely arbitrary decision on the minimum dimensions...Preferences on the minimum, @pyrello?

Hmm... looks like the image wasn't set up as responsive. Need to look into that quick. My initial idea would be nothing with a width below 300px (height shouldn't matter because any width that meets that requirement should be fine on height).

> Also, currently if it doesn't reach the minimum it won't be pulled in, but that can be updated if the idea was to keep the smaller image but list it as one to be replaced later. Realized I'd misread the original comment.

I think we want it to pull the image for now and generate a list for follow-up afterwards. Otherwise they can't launch until that clean-up work is done.

**Comment by cory-skeers** [comment]
> I think we want it to pull the image for now and generate a list for follow-up afterwards. Otherwise they can't launch until that clean-up work is done.

Updated it to still pull the image, but give a post-migrate log of node: image that can be used for follow-up. I also set it to use 300 pixels as the threshold for width, and to not care about height. Since the threshold is used for informational purposes only, the exact pixel value isn't as crucial, so I think the 300 initial thought is adequate (though it's a minor update if needed).

I'd also set up the redirects migration since it's so straightforward to do, but if there's any reason to specifically avoid it, we can tear it out.
